src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/ComickFunDto.kt

```diff
@@ -48,7 +49,15 @@ data class Manga(
         thumbnail_url = parseCover(comic.cover, comic.mdCovers)
         artist = artists.joinToString { it.name.trim() }
         author = authors.joinToString { it.name.trim() }
-        genre = (listOfNotNull(comic.origination) + genres)
+        genre = listOf(
+            listOfNotNull(comic.origination),
+            listOf(Name(demographic.orEmpty())),
+            genres,
+            comic.mdGenres.mapNotNull { it.name },
+            if (muTags) comic.muGenres.categories.map { Name(it.category?.title.orEmpty()) } else emptyList(),
+        )
+            .flatten()
+            .distinct()
             .joinToString { it.name.trim() }
```

Comment:
I'm not sure if we actually need those blank checks
```suggestion
        genre = buildList {
            comic.origination?.let(::add)
            demographic?.takeUnless(String::isBlank)?.let { add(Name(it)) }
            addAll(genres)
            addAll(comic.mdGenres.mapNotNull { it.name })
            if (muTags) {
                comic.muGenres.categories.forEach { category ->
                    category.category?.title?.takeUnless(String::isBlank)?.let { add(Name(it)) }
                }
            }
        }
            .distinctBy { it.name }
            .joinToString { it.name.trim() }
```

Reply:
Maybe replace them with a `.filter { it.name.isNotBlank() }` before the string conversion?